libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -191,16 +193,22 @@ public XContentBuilder(XContent xContent, OutputStream bos, Set<String> includes
      * remaining fields against the inclusive filters.
      * <p>
      * Make sure to call {@link #close()} when the builder is done with.
-     *
-     * @param os       the output stream
+     *  @param os       the output stream
      * @param includes the inclusive filters: only fields and objects that match the inclusive filters will be written to the output.
      * @param excludes the exclusive filters: only fields and objects that don't match the exclusive filters will be written to the output.
+     * @param responseContentTypeString  a content-type header value to be send back on a response
      */
-    public XContentBuilder(XContent xContent, OutputStream os, Set<String> includes, Set<String> excludes) throws IOException {
+    public XContentBuilder(XContent xContent, OutputStream os, Set<String> includes, Set<String> excludes,
```

Comment:
something to consider for later ... but I wonder if there is any overlap between the responseContentTypeString set here and ParsedMediaType. Meaning that ParsedMediaType is essentially  String -> Parts... I wonder if there is the opposite where we could pass a ResponseMediaType that would be Parts -> String. It would clean up the contract abit since String as parameters is a fairly loose contract.

Reply:
I think this is a good idea. The problem is thought that in AbstractRestChannel where we create XContentBuilder https://github.com/elastic/elasticsearch/pull/65500/files#diff-05da46b91fb0631c2c11e6aba2b58fa150162cf6c55bcd9bd5751b704090e8d8R137 ResponseContentType can be created in different ways. 
We would have to create an XContentType first (in one case it is created from ParsedMediaType), then create ParsedMediaType out of it.

I think overall it is still better then just a string.
updated the PR. let me know what you think

Reply:
Looks good, thanks for the change. Much better contract than String.